### PR TITLE
Allow mutliple API flags in matc

### PR DESCRIPTION
**Pull request by bejado**
This allows specifying individual APIs on the command line / with `filamat`:

```
matc --api opengl --api metal -o test material.mat
```

`TargetApi` is now a bitmask. Clients code should not need to change.

This is part 1 of #1314

### CODE REVIEW

File: libs/filamat/src/PostprocessMaterialBuilder.cpp [hunk 1]
```diff
@@ -40,6 +40,12 @@ using namespace filament::backend;
 
 namespace filamat {
 
+inline void assertSingleTargetApi(MaterialBuilderBase::TargetApi api) {
```
Comment (bejado) [inline libs/filamat/src/PostprocessMaterialBuilder.cpp]: This is duplicated here, but `PostprocessMaterialBuilder` will be deprecated soon anyway

File: libs/filamat/include/filamat/MaterialBuilder.h [hunk 2]
```diff
@@ -49,11 +49,11 @@ class UTILS_PUBLIC MaterialBuilderBase {
         ALL
     };
 
-    enum class TargetApi {
-        ALL,
-        OPENGL,
-        VULKAN,
-        METAL
+    enum class TargetApi : uint8_t {
+        OPENGL      = 0x01,
```
Comment (romainguy) [inline libs/filamat/include/filamat/MaterialBuilder.h]: You must also update the Java bindings.

File: libs/filamat/src/MaterialBuilder.cpp [hunk 3]
```diff
@@ -54,6 +54,12 @@ namespace filamat {
 
 std::atomic<int> MaterialBuilderBase::materialBuilderClients(0);
 
+inline void assertSingleTargetApi(MaterialBuilderBase::TargetApi api) {
```
Comment (prideout) [inline libs/filamat/src/MaterialBuilder.cpp]: Ash thanks for adding this assert! My personal aesthetic would've been to use strong typing instead, but this works.

File: libs/filamat/src/MaterialBuilder.cpp [hunk 4]
```diff
@@ -293,7 +298,7 @@ MaterialBuilder& MaterialBuilder::platform(Platform platform) noexcept {
 }
 
 MaterialBuilder& MaterialBuilder::targetApi(TargetApi targetApi) noexcept {
-    mTargetApi = targetApi;
+    mTargetApi |= targetApi;
```
Comment (romainguy) [inline libs/filamat/src/MaterialBuilder.cpp]: You should comment this behavior in the .h.

**Pull request merged**